rq/job.py

```diff
@@ -561,9 +570,26 @@ def delete(self, pipeline=None, remove_from_queue=True):
                                             job_class=self.__class__)
             failed_queue.remove(self, pipeline=pipeline)
 
+        if delete_dependents:
+            self.delete_dependents(pipeline=pipeline,
+                                   remove_from_queue=remove_from_queue)
+
         connection.delete(self.key)
         connection.delete(self.dependents_key)
 
+    def delete_dependents(self, pipeline=None, remove_from_queue=True):
+        """Delete jobs depending on this job."""
+        connection = pipeline if pipeline is not None else self.connection
+        for dependent_id in self.dependent_ids:
+            try:
+                job = Job.fetch(dependent_id, connection=self.connection)
+                job.delete(pipeline=pipeline,
+                           remove_from_queue=remove_from_queue)
```

Comment:
Please use `remove_from_queue=False` for this one so RQ doesn't try to remove the job from Redis (we save an extra Redis operation here).